hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/SingleRecipientMessageSenderAdapter.java

```diff
@@ -0,0 +1,32 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender;
+
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+import pl.allegro.tech.hermes.consumers.consumer.SendFutureProvider;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+public class SingleRecipientMessageSenderAdapter implements MessageSender {
```

Comment:
The idea was for it be an umbrella class for all message senders with only one recipient so that rate limiting would only be done here and in `JettyBroadcastMessageSender`. Moving rate limiting to each implementee would result in many duplicated tests for rate limiting.